packages/output/src/common/output-channel.ts

```diff
@@ -217,10 +213,13 @@ export class OutputChannelManager implements CommandContribution, Disposable, Re
         const { name } = channel;
         return new DisposableCollection(
             channel,
-            channel.onVisibilityChange(({ isVisible }) => {
+            channel.onVisibilityChange(async ({ isVisible, preserveFocus }) => {
                 if (isVisible) {
                     this.selectedChannel = channel;
-                    this.channelWasShownEmitter.fire({ name });
+                    const reveal = true;
+                    const activate = !preserveFocus;
+                    await this.outputContribution.openView({ reveal, activate });
```

Comment:
i think the original idea was that channel represents the state which the contribution should listen to and react. The channel should depend on concrete views.

Reply:
OK, this makes sense. 👍